lib/opal/compiler.rb

```diff
@@ -253,32 +254,32 @@ def unique_temp(name)
       name = name.to_s
       if name && !name.empty?
         name = "_#{name}"
-          .gsub('?', '$q')
-          .gsub('!', '$B')
-          .gsub('=', '$eq')
-          .gsub('<', '$lt')
-          .gsub('>', '$gt')
-          .gsub(/[^\w\$]/, '$')
+               .gsub('?', '$q')
+               .gsub('!', '$B')
+               .gsub('=', '$eq')
+               .gsub('<', '$lt')
+               .gsub('>', '$gt')
+               .gsub(/[^\w\$]/, '$')
       end
       unique = (@unique += 1)
       "TMP#{name}_#{unique}"
     end
 
     # Use the given helper
     def helper(name)
-      self.helpers << name
+      helpers << name
     end
 
     # To keep code blocks nicely indented, this will yield a block after
     # adding an extra layer of indent, and then returning the resulting
     # code after reverting the indent.
-    def indent(&block)
+    def indent
```

Comment:
I tend to prefer explicit over implicit

Reply:
...And we come to the question about the definition of "implicit". I personally think that `yield` is implicit as methods may have only 1 block that gets automatically required and called (i.e. if there's no block it throws a `LocalJumpError: no block given`). So I interpret `yield` is like a macro or something similar.
@elia what do you think? I'm not against reverting it, we can just disable this rule, revert what I've changed (too keep history clean) and allow everyone to choose how to call blocks.

Reply:
From what I remember in MRI yield is faster, so probably when a block is named `block` and is not required to be an object (to be stored or forwarded to another method etc.) it probably make sense to just use `yield`.

If the block is named differently probably the name helps in telling the story of what's going on and should be preserved.

The other advantage of having the block among the arguments is that you know immediately that a block is expected by that method and you don't have to check every line of the implementation parsing for `yield`.

Reverting the rule is fine, mostly because I don't think rubocop can act based on the block variable name.